third_party/nvfuser/csrc/python_frontend/fusion_definition.cpp

```diff
@@ -123,6 +127,41 @@ void FusionDefinition::finalizeDefinition() {
   }
 }
 
+void FusionDefinition::setupSchedule(const at::ArrayRef<c10::IValue>& inputs) {
+  FUSER_PERF_SCOPE("FusionDefinition::setupSchedule");
+  TORCH_CHECK(
+      fusion_id_.has_value(), "FusionDefinition definition does not exist!");
+  auto& scheds = fusionCache()->queryFusionSchedules(fusion_id_.value());
```

Comment:
So fusionCache gives us the singleton.. Which means any access to that needs to be thread-safe...

Reply:
IN CASE THIS COMMENT IS EASILY OVERLOOKED, I'M PUTTING AN ALL CAP COMMENT BELOW IT TO ENSURE IT'S GETTING THE ATTENTION IT DESERVES. LOL

Reply:
I fixed up some of thread-safety by locking the `queryFusionSchedules()`, `queryChildren()`, and `createChild()`.  However, this does not go far enough.  I discussed what needs to be addressed in a future PR above in the main comment for this PR.